File: 백준/Gold/14719. 빗물/빗물.cc
//백준 14719, 빗물

#include <iostream>

int arr[501];

int main (){

    int H, W;
    std::cin >> H >> W;

    for(int i{0}; i<W; ++i){
        std::cin >> arr[i];
    }

    int cnt{0};
    for(int i{1}; i<W-1; ++i){
        int l{0}, r{0};
        
        for(int j{0}; j<i; ++j) l = std::max(l, arr[j]);
        for(int j{W-1}; j>i; --j) r = std::max(r, arr[j]);
        
        cnt += std::max(0, std::min(l, r)-arr[i]);
    }
    
    std::cout << cnt;

    return 0;
}

File: 백준/Gold/14719. 빗물/빗물.py
H, W = map(int, input().split())
blocks = list(map(int, input().split()))

cnt = 0

for i in range(1, W-1):
  left = 0; right = 0
  for j in range(0, i):
    left = max(left, blocks[j])
  for j in range(W-1, i, -1):
    right = max(right, blocks[j])
  
  cnt += max(0, min(left, right)-blocks[i])

print(cnt)